packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -982,6 +1010,16 @@ export interface IBuildImage {
    */
   readonly defaultComputeType: ComputeType;
 
+  /**
+   * The type of credentials AWS CodeBuild uses to pull images in your build.
+   */
+  readonly imagePullCredentialsType?: ImagePullCredentialsType;
```

Comment:
It's important that we specify what is the default behavior here.

Reply:
This is an attribute in `IBuildImage` and customers cannot explicitly specify it. It will only be used by `fromEcrRepository`, `fromAsset` and `fromDockerRegistry`. Does it still make sense to add the `@default` tag here as customers don't have control over it?

Reply:
I think it's still useful. We do foresee customers implementing this interface directly (for example, when working around a feature gap in our existing implementations, `LinuxBuildImage` and `WindowsBuildImage`), so I would say documenting the default behavior is important here.